src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptions.cs

```diff
@@ -90,6 +91,26 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 context.Pipeline.Register(new SendOnlySubscribeTerminator(), "Throws an exception when trying to subscribe from a send-only endpoint");
                 context.Pipeline.Register(new SendOnlyUnsubscribeTerminator(), "Throws an exception when trying to unsubscribe from a send-only endpoint");
             }
+
+            context.Container.ConfigureComponent<CallInit>(DependencyLifecycle.SingleInstance);
+
+            context.RegisterStartupTask(b => b.Build<CallInit>());
+        }
+
+        class CallInit : FeatureStartupTask
+        {
+            public IInitializableSubscriptionStorage SubscriptionStorage { get; set; }
```

Comment:
so we're still using property injection here? :(

Reply:
As per discussion on Friday, we still need to